skyrim-platform/src/platform_se/skyrim_platform/GameEventSinks.cpp

```diff
@@ -1681,7 +1686,28 @@ RE::BSEventNotifyControl GameEventSinks::ProcessEvent(
   auto refId = ref ? ref->formID : 0;
   auto type = converted->type;
 
-  SkyrimPlatform::GetSingleton().AddUpdateTask([ref, refId, type] {
+  const char* eventName;
+  switch (type) {
+    case TESEvents::TESObjectREFRTranslationEvent::EventType::kFailed: {
+      eventName = "translationFailed";
+      break;
+    }
+    case TESEvents::TESObjectREFRTranslationEvent::EventType::
+      kAlmostCompleted: {
+      eventName = "translationAlmostCompleted";
+      break;
+    }
+    case TESEvents::TESObjectREFRTranslationEvent::EventType::kCompleted: {
+      eventName = "translationCompleted";
+      break;
+    }
+  }
+
+  if (!EventsApi::HasSubscriptions(eventName)) {
```

Comment:
It's not guaranteed in which thread event sink is fired so we cannot just use `HasSubscriptions` here and in any other place of event sink handlers. It's because `HasSubscroptions` provides unsynchronized access to subscriptions.

Even if we sync access I think it would be expensive.

We can probably create 4 `atomic_flags` for events (hardcode these events).
https://en.cppreference.com/w/cpp/atomic/atomic_flag

And **do not fire process events until one subscribes**.
Once someone subscribes we can say **OK now I go unoptimized**